packages/recommendations-react/src/useAlgoliaAgent.ts

```diff
@@ -0,0 +1,12 @@
+import { GetRelatedProductsProps } from '@algolia/recommendations-core';
+import { useEffect } from 'react';
+
+import { version } from './version';
+
+type UseAlgoliaAgentProps = Pick<GetRelatedProductsProps<any>, 'searchClient'>;
+
+export function useAlgoliaAgent(props: UseAlgoliaAgentProps) {
+  useEffect(() => {
```

Comment:
in theory this code could also be shared between vdom implementations, but then you'll need to map to exactly useEffect in Vue, which might not be straightforward, so copied makes sense

Reply:
Yeah, here we want to avoid to use framework specifics and share them across packages, otherwise we go back to the original building/aliasing issues easily. This "code duplication" will be counter-balance with some testing.